stl/inc/valarray

```diff
@@ -195,99 +184,175 @@ public:
     valarray& operator=(const indirect_array<_Ty>& _Indarr); // defined below
 
     _NODISCARD valarray operator+() const {
-        _VALOP(_Ty, size(), +_Myptr[_Idx]);
+        valarray<_Ty> _Ans(size());
+        for (size_t _Idx = 0; _Idx < _Ans.size(); ++_Idx) {
+            _Ans[_Idx] = +_Myptr[_Idx];

```

Comment:
I believe the `+` is unnecessary

Reply:
Are you sure that for any `NumericType` unary + is no operation?

Reply:
Urgh, the only thing I could imagine was integer promotion, but that should not be any problem as we asign the numeric type anyway and there cannot be any overflow.

Reply:
Nothing special can happen with normal integers or floats.

Assume some weird type where unary + does some normalization. Lambda is quite close example, although its + causes type conversion.

Reply:
And Numeric Type is defined by `valarray` quite broad. `std::string` is a Numeric Type, for example